and-bible/app/src/main/java/net/bible/android/control/backup/BackupControl.java

```diff
@@ -76,9 +80,13 @@ public void okay() {
 		});
 	}
 	
-	/** return true if a backup has been done and the file is on the sd card
+	/** return true if a backup has been done and the file is on the sd card. Also return true if permission is not yet
+	* granted and we can't know if there is a file (it will be requested)
 	 */
-	private boolean isBackupFile() {
+	private boolean isBackupEnabled() {
+		if(ContextCompat.checkSelfPermission(BibleApplication.getApplication(), Manifest.permission.READ_EXTERNAL_STORAGE) == PackageManager.PERMISSION_DENIED) {
+			return true;

```

Comment:
The purpose of this method is to enable the 'Restore' menu item only if a backup exists.  If a backup does not exist the Restore menu item is disabled.

Therefore, this should be 
    return false;
if the app does not have read permission then it also will not have write permission (write perm includes read perm) which means the app could not have saved a backup.

BUT ACTUALLY a backup could exist is if the user manually copied a file to the backup directory.  That could occur if the user has switched mobile phones and wishes to transfer data.  So maybe the solution is to remove this method totally and to always enable the restore backup menu item.  If the user requests a Restore then READ permission will have to be checked and possibly requested.  After READ permission has been granted then a message that no backup file exists can be shown ot the user if there is no file.

Reply:
I think that the function now works correctly. 

- if we don't have permission, backup file could exist but we don't know. (it will request permission when using function)

- if we have permission, we know if backup file exists or not and thus we can enable/disable menu entry right away.